packages/loader/container-loader/src/lifecycleStateHandler.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+
+type LifecycleState = "created" | "loading" | "loaded" | "closing" | "closed";
+
+export class LifecycleStateHandler {
+    public get state(): LifecycleState {
+        return this._state;
+    }
+
+    /**
+     * Returns true if the container has been closed, otherwise false
+     */
+    public isClosed(): boolean {
```

Comment:
I'd rather use names that are reflecting actual state. Here name hides the fact that it can be in closing state, and that's not fully abstracted away (as raw state is exposed)